frontend/src/pages/voting/index.tsx

```diff
@@ -130,108 +132,127 @@ export const VotingPage = () => {
+                    sx={{
+                      background: "orange",
+                      borderRadius: 0,
+                    }}
+                  >
+                    <FormattedMessage id="voting.allTopics">
+                      {(text) => <option value="">{text}</option>}
+                    </FormattedMessage>
+                    {data?.conference.topics.map((topic) => (
+                      <option key={topic.id} value={topic.id}>
+                        {topic.name}
+                      </option>
+                    ))}
+                  </Select>
+
+                  <Select
+                    {...select("language")}
+                    sx={{
+                      background: "violet",
+                      borderRadius: 0,
+                    }}
+                  >
+                    <FormattedMessage id="voting.allLanguages">
+                      {(text) => <option value="">{text}</option>}
+                    </FormattedMessage>
+                    {data?.conference.languages.map((language) => (
+                      <option key={language.id} value={language.code}>
+                        {language.name}
+                      </option>
+                    ))}
+                  </Select>
+
+                  <Select
+                    {...select("vote")}
+                    sx={{
+                      borderRadius: 0,
+                    }}
+                  >
+                    <FormattedMessage id="voting.allSubmissions">
+                      {(text) => <option value="all">{text}</option>}
+                    </FormattedMessage>
+                    <FormattedMessage id="voting.notVoted">
+                      {(text) => <option value="notVoted">{text}</option>}
+                    </FormattedMessage>
+                    <FormattedMessage id="voting.votedOnly">
+                      {(text) => <option value="votedOnly">{text}</option>}
+                    </FormattedMessage>
+                  </Select>
+
+                  <TagsFilter
+                    {...raw("tags")}
+                    tags={data?.submissionTags ?? []}
+                  />
+                </Grid>
+              )}
           </Grid>
         </Box>
       </Box>
 
-      {loggedIn && (loading || cannotVoteErrors || error) && (
-        <Box
-          sx={{
-            maxWidth: "container",
-            mx: "auto",
-            px: 3,
-          }}
-        >
-          {!cannotVoteErrors && error && (
-            <Alert variant="alert">{error.message}</Alert>
-          )}
-
-          {cannotVoteErrors && error && (
-            <Alert variant="alert">
-              <Link path="/tickets">
-                <FormattedMessage id="voting.buyTicketToVote" />
-              </Link>
-            </Alert>
-          )}
-          {loading && (
-            <Alert variant="info">
-              <FormattedMessage id="voting.loading" />
-            </Alert>
-          )}
+      {userCannotVote && (
```

Comment:
Uhm this looks a bit weird to me / complex, the nesting is not ideal also I don't think we should show a "You can't vote" while the network is loading (this is what happens right?), it is very confusing

Reply:
Nope when the network is loading we will show Loading, because the above condition at line 230